From 0b1deb9883ad8776f8774ab3d4e617c4823c26b2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:09 -0400
Subject: [powerpc] clean-up include ordering in prom.h

Message-id: <1410545655-205645-81-git-send-email-dzickus@redhat.com>
Patchwork-id: 94141
O-Subject: [RHEL7 PATCH 080/626] powerpc: clean-up include ordering in prom.h
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5c19c5c6d4f5e1dc0d0e26b683bc820dda01fe06
Author: Rob Herring <rob.herring@calxeda.com>
Date:   Tue Sep 17 14:34:00 2013 -0500

    powerpc: clean-up include ordering in prom.h

    Now that the core OF headers don't depend on prom.h, rearrange the
    includes. There are still lots of implicit includes in the powerpc tree,
    so the includes of OF headers are still necessary.

    Signed-off-by: Rob Herring <rob.herring@calxeda.com>
    Acked-by: Grant Likely <grant.likely@linaro.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/prom.h b/arch/powerpc/include/asm/prom.h
index e6ec2cf..1d4a021 100644
--- a/arch/powerpc/include/asm/prom.h
+++ b/arch/powerpc/include/asm/prom.h
@@ -1,4 +1,3 @@
-#include <linux/of.h> /* linux/of.h gets to determine #include ordering */
 #ifndef _POWERPC_PROM_H
 #define _POWERPC_PROM_H
 #ifdef __KERNEL__
@@ -22,6 +21,13 @@
 
 #define HAVE_ARCH_DEVTREE_FIXUPS
 
+/* These includes should be removed once implicit includes are cleaned up. */
+#include <linux/of.h>
+#include <linux/of_fdt.h>
+#include <linux/of_address.h>
+#include <linux/of_irq.h>
+#include <linux/platform_device.h>
+
 /*
  * OF address retreival & translation
  */
@@ -146,14 +152,5 @@ struct of_drconf_cell {
  */
 extern unsigned char ibm_architecture_vec[];
 
-/* These includes are put at the bottom because they may contain things
- * that are overridden by this file.  Ideally they shouldn't be included
- * by this file, but there are a bunch of .c files that currently depend
- * on it.  Eventually they will be cleaned up. */
-#include <linux/of_fdt.h>
-#include <linux/of_address.h>
-#include <linux/of_irq.h>
-#include <linux/platform_device.h>
-
 #endif /* __KERNEL__ */
 #endif /* _POWERPC_PROM_H */
-- 
1.7.1